reporter/internal/pdata/generate.go

```diff
@@ -239,31 +246,33 @@ func (p *Pdata) setProfile(
 
 	// Populate the deduplicated functions into profile.
 	funcTable := dic.FunctionTable()
-	funcTable.EnsureCapacity(len(funcMap))
+	funcTable.EnsureCapacity(len(funcMap) + int(funcDicOffset))
 	for range funcMap {
 		funcTable.AppendEmpty()
 	}
 	for v, idx := range funcMap {
 		f := funcTable.At(int(idx))
-		f.SetNameStrindex(getStringMapIndex(stringMap, v.Name))
-		f.SetFilenameStrindex(getStringMapIndex(stringMap, v.FileName))
+		f.SetNameStrindex(getStringMapIndex(stringMap, strDicOffset, v.Name))
+		f.SetFilenameStrindex(getStringMapIndex(stringMap, strDicOffset, v.FileName))
 	}
 
 	// When ranging over stringMap, the order will be according to the
 	// hash value of the key. To get the correct order for profile.StringTable,
 	// put the values in stringMap, in the correct array order.
-	stringTable := make([]string, len(stringMap))
+	tmpStringTable := make([]string, len(stringMap)+int(strDicOffset))
```

Comment:
Yes - this can become a massive waste of memory, as `strDicOffset` of elements are just empty elements ...
`dic.StringTable()` is missing a functionality like `AppendEmpty()`, as `EnsureCapacity()` capacity is not sufficient to grow the underlying data structure.